enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+---
+title: assisted-installer-in-cluster
+authors:
+  - "@mhrivnak"
+  - "@hardys"
+  - "@avishayt"
+reviewers:
+  - "@eparis"
+  - "@markmc"
+  - "@dgoodwin"
+  - "@ronniel1"
+
+approvers:
+  - TBD
+creation-date: 2020-12-22
+last-updated: 2020-12-22
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+replaces:
+superseded-by:
+---
+
+# Assisted Installer in-cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' clusters for two purposes:
+
+1. A cluster created by the Assisted Installer needs the ability to add workers
+on day 2.
+2. Multi-cluster management, such as through hive and ACM, should be
+able to utilize the capabilities of the Assisted Installer.
+
+This enhancement proposes the first iteration of running the Assisted Installer
+in end-user clusters to meet the purposes above.
+
+## Motivation
+
+### Goals
+
+* Expose the Assisted Installer's capabilities as a kubernetes-native API.
```

Comment:
OpenShift's infrastructure management is k8s-native, so agent-based installation through assisted installer needs to be exposed as k8s-native APIs. That's a design principle of OpenShift 4 that is outside the scope of this enhancement, though it wouldn't hurt to re-state it here.

As for hive, hive is not necessarily the only user of these APIs, so it would not be a suitable point at which to bridge the APIs from non-native to native.

Reply:
Related to comment I just added below, but I think it's important to remember what Hive is. Hive is not a big bloated management application, it's an API for installing OpenShift at scale. I am presently thinking we should work together to get this into Hive and eliminate this middle API + operator.